cou/utils/openstack.py

```diff
@@ -147,3 +230,98 @@ def lookup(cls, component: str, version: str) -> List[str]:
             if version in version_range:
                 compatible_os_releases.append(openstack_release)
         return compatible_os_releases
+
+
+class OpenStackRelease:
+    """Provides a class that will compare strings from an iterator type object.
+
+    Used to provide > and < comparisons on strings that may not necessarily be
+    alphanumerically ordered.  e.g. OpenStack or Ubuntu releases AFTER the
+    z-wrap.
+    """
+
+    openstack_codenames = list(OPENSTACK_CODENAMES.keys())
+    openstack_release_year = list(OPENSTACK_CODENAMES.values())
+
+    def __init__(self, os_release: str):
+        """Initialize the OpenStackRelease object.
+
+        :param os_release: OpenStack release codename.
+        :type os_release: str
+        :raises ValueError: Raises ValueError if OpenStack codename is unknown.
+        """
+        self.os_release = os_release
+        try:
+            self.index = self.openstack_codenames.index(os_release)
+        except ValueError as exc:
+            raise ValueError(
+                f"OpenStack '{os_release}' is not in '{self.openstack_codenames}'"
+            ) from exc
+
+    def __eq__(self, other):
+        """Do equals."""
+        assert isinstance(other, (str, self.__class__))
```

Comment:
This does not make sense to me, if I do `OpenStackRelease("yoga") == None` it raise AssertError.
```python
def __eq__(self, other):
    if not isinstance(other, Partial):
        return NotImplemented

    return self.index == other.index
```

Reply:
And you did not even test this behavior with any unit tests.

Reply:
If we just use the index to compare it won't be possible to compare a string with the ` OpenStackRelease` object. I think it's nice to be able to do something like:

```python
OpenStackRelease("victoria") > "ussuri" 
```

Using this approach just comparing `OpenStackRelease` will be possible.

Reply:
Sure, I'm returning NotImplemented, but I kept the comparison.